feature-staking-api/src/main/java/io/novafoundation/nova/feature_staking_api/data/proxy/AddStakingProxyRepository.kt

```diff
@@ -0,0 +1,16 @@
+package io.novafoundation.nova.feature_staking_api.data.proxy
+
+import io.novafoundation.nova.feature_account_api.data.extrinsic.ExtrinsicSubmission
+import io.novafoundation.nova.feature_account_api.data.model.Fee
+import io.novafoundation.nova.feature_proxy_api.domain.model.ProxyDepositWithQuantity
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+
+interface AddStakingProxyRepository {
```

Comment:
We usually do not have a separate repository for fee and action, we just directly use `extrinsicService` + extrinsic builder's extension. What is the need for it here? Also, why do we have it in the API module? I cannot quickly imagine a usecase when this interface might be useful in other modules

Reply:
Changed repository to interactor and moved it to impl module only